src/EnergyPlus/Material.hh

```diff
@@ -145,31 +161,58 @@ namespace Material {
     constexpr std::array<std::string_view, static_cast<int>(VariableAbsCtrlSignal::Num)> VariableAbsCtrlSignalUC = {
         "SURFACETEMPERATURE", "SURFACERECEIVEDSOLARRADIATION", "SPACEHEATINGCOOLINGMODE", "SCHEDULED"};
 
-    struct MaterialProperties
+    // Parameters to indicate surface roughness for use with the Material
+    // derived type:
+    enum class SurfaceRoughness
+    {
+        Invalid = -1,
+        VeryRough,
+        Rough,
+        MediumRough,
+        MediumSmooth,
+        Smooth,
+        VerySmooth,
+        Num
+    };
+
+    constexpr std::array<std::string_view, static_cast<int>(SurfaceRoughness::Num)> SurfaceRoughnessUC{
+        "VERYROUGH", "ROUGH", "MEDIUMROUGH", "MEDIUMSMOOTH", "SMOOTH", "VERYSMOOTH"};
+
+    extern const std::array<std::string_view, static_cast<int>(SurfaceRoughness::Num)> RoughnessNames;
+
+    struct MaterialBase
     {
 
         // Members
-        std::string Name = "";                                            // Name of material layer
-        Material::MaterialGroup Group = Material::MaterialGroup::Invalid; // Material group type (see Material Parameters above.  Currently
+        std::string Name = "";        // Name of material layer
+        Group group = Group::Invalid; // Material group type (see Material Parameters above.  Currently
         // active: RegularMaterial, Shade, Air, WindowGlass,
         // WindowGas, WindowBlind, WindowGasMixture, Screen, EcoRoof,
         // IRTMaterial, WindowSimpleGlazing, ComplexWindowShade, ComplexWindowGap)
-        DataSurfaces::SurfaceRoughness Roughness =
-            DataSurfaces::SurfaceRoughness::Invalid; // Surface roughness index (See Surface Roughness parameters
+        SurfaceRoughness Roughness = SurfaceRoughness::Invalid; // Surface roughness index (See Surface Roughness parameters
         // above.  Current: VerySmooth, Smooth, MediumSmooth,
         // MediumRough, Rough, VeryRough)
         // Thermo-physical material properties
         Real64 Conductivity = 0.0; // Thermal conductivity of layer (W/m2K)
         Real64 Density = 0.0;      // Layer density (kg/m3)
-        Real64 IsoMoistCap = 0.0;  // Isothermal moisture capacity on water vapor density (m3/kg)
-        Real64 Porosity = 0.0;     // Layer porosity
         Real64 Resistance = 0.0;   // Layer thermal resistance (alternative to Density,
         // Conductivity, Thickness, and Specific Heat; K/W)
-        bool ROnly = false;         // Material defined with "R" only
-        Real64 SpecHeat = 0.0;      // Layer specific heat (J/kgK)
+        bool ROnly = false;     // Material defined with "R" only
+        Real64 SpecHeat = 0.0;  // Layer specific heat (J/kgK)
+        Real64 Thickness = 0.0; // Layer thickness (m)
+
+        virtual bool dummy()
+        {
+            return true;
+        }
+    };
+
+    struct MaterialChild : public MaterialBase
```

Comment:
MaterialChild feels like the wrong name.  If we are going to use class inheritance, the names should be more meaningful.  

```c++
struct MaterialBase {};
struct MaterialRegular : MaterialBase {};
struct MaterialSomething : MaterialBase {};
```

Reply:
This is the second step of splitting. After this, we'll subdivide MaterialChild into more meaningful classes like gas/gas mixture, simple material, complex opaque, complex window, potentially some other types

Reply:
Yep, that's a great path forward.